addons/medical_engine/functions/fnc_handleDamage.sqf

```diff
@@ -38,13 +38,37 @@ private _newDamage = _damage - _oldDamage;
 // Get armor value of hitpoint and calculate damage before armor
 private _armor = [_unit, _hitpoint] call FUNC(getHitpointArmor);
 private _realDamage = _newDamage * _armor;
-// Damages are stored for "ace_hdbracket" event triggered last
-_unit setVariable [format [QGVAR($%1), _hitPoint], [_realDamage, _newDamage]];
 TRACE_3("Received hit",_hitpoint,_newDamage,_realDamage);
 
-// Engine damage to these hitpoints controls blood visuals, limping, weapon sway
-// Handled in fnc_damageBodyPart, persist here
-if (_hitPoint in ["hithead", "hitbody", "hithands", "hitlegs"]) exitWith {_oldDamage};
+// Drowning doesn't fire the EH for each hitpoint so the "ace_hdbracket" code never runs
+// Damage occurs in consistent increments
+if (
+    _hitPoint isEqualTo "#structural" &&
+    {getOxygenRemaining _unit <= 0.5} &&
+    {_damage isEqualTo (_oldDamage + 0.005)}
+) exitWith {
+    TRACE_5("Drowning",_unit,_shooter,_instigator,_damage,_newDamage);
+    [QEGVAR(medical,woundReceived), [_unit, [[_newDamage, "Body", _newDamage]], _unit, "drowning"]] call CBA_fnc_localEvent;
+
+    0
+};
+
+// Crashing a vehicle doesn't fire the EH for each hitpoint so the "ace_hdbracket" code never runs
+// It does fire the EH multiple times, but this seems to scale with the intensity of the crash
+private _vehicle = vehicle _unit;
+if (
+    EGVAR(medical,enableVehicleCrashes) &&
+    {_hitPoint isEqualTo "#structural"} &&
+    {_ammo isEqualTo ""} &&
+    {_vehicle != _unit} &&
+    {vectorMagnitude (velocity _vehicle) > 5}
+    // todo: no way to detect if stationary and another vehicle hits you
+) exitWith {
+    TRACE_5("Crash",_unit,_shooter,_instigator,_damage,_newDamage);
+    [QEGVAR(medical,woundReceived), [_unit, [[_newDamage, _hitPoint, _newDamage]], _unit, "vehiclecrash"]] call CBA_fnc_localEvent;
+
+    0
+};
```

Comment:
These "special" damage sources are now evaluated first so their damage doesn't get saved to the unit - previously it was being saved and not cleared so it would be included in the next event that triggered `ace_hdbracket`. It didn't cause any problems because it's only `#structural` damage which isn't used anywhere, but could have in the future.

Saving of damage is now done last, so it gets skipped any time wounds are actually applied.